test_registry.py

```diff
@@ -0,0 +1,24 @@
+import os
+import unittest
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class ToxDockerRegistryTest(unittest.TestCase):
+    def test_it_sets_specific_env_vars(self):
+        self.assertEqual("env-var-value", os.environ["ENV_VAR"])
+
+    def test_it_sets_automatic_env_vars(self):
+        # the nginx image we use exposes port 80
+        self.assertIn("DOCKER_IO_LIBRARY_NGINX_HOST", os.environ)
+        self.assertIn("DOCKER_IO_LIBRARY_NGINX_80_TCP", os.environ)
```

Comment:
It's not clear to me that this should be what you have here vs. eg `NGINX_80_TCP`. What's the use case for having multiple containers with the same name running (if they come from different regstries)?

@tox-dev/maintainers any opinions?

Reply:
Sorry for the delay, I miss the notification for your replies.
The aim is to replace all invalid characters in the environment variable name.
Without the fix, an image from `docker.io/library/nginx:1.13-alpine` (or another private registry) will create an env var with the name `DOCKER.IO_LIBRARY_NGINX_80_TCP` which is not valid for a environment var.